src/tools/ts-generator/linksTSgenerator.cpp

```diff
@@ -0,0 +1,346 @@
+
+#include "linksTSgenerator.h"
+#include <yuni/datetime/timestamp.h>
+
+namespace Antares::TSGenerator
+{
+// ==================
+// Free functions
+// ==================
+std::vector<std::string> extractTargetAreas(fs::path sourceLinkDir)
+{
+    std::vector<std::string> to_return;
+    fs::path pathToIni = sourceLinkDir / "properties.ini";
+    IniFile ini;
+    ini.open(pathToIni); // gp : we should handle reading issues
+    for (auto* s = ini.firstSection; s; s = s->next)
+    {
+        std::string targetAreaName = transformNameIntoID(s->name);
+        to_return.push_back(targetAreaName);
+    }
+    return to_return;
+}
+
+bool pairs_match(const LinkPair& p1, const LinkPair& p2)
+{
+    return (p1.first == p2.first && p1.second == p2.second)
+           || (p1.first == p2.second && p1.second == p2.first);
+}
+
+const LinkPair* getMatchingPairInCollection(const LinkPair& pair, const LinkPairs& collection)
+{
+    for(const auto& p : collection)
+    {
+        if (pairs_match(pair, p))
+            return &p;
+    }
+    return nullptr;
+}
+
+bool readLinkGeneralProperty(StudyParamsForLinkTS& params,
+                             const Yuni::String& key,
+                             const Yuni::String& value)
+{
+    if (key == "derated")
+    {
+        return value.to<bool>(params.derated);
+    }
+    if (key == "nbtimeserieslinks")
+    {
+        return value.to<unsigned int>(params.nbLinkTStoGenerate);
+    }
+    if (key == "seed-tsgen-links")
+    {
+        unsigned int seed {0};
+        if (! value.to<unsigned int>(seed))
+            return false;
+        params.random.reset(seed);
+        return true;
+    }
+    return true; // gp : should we return true here ?
```

Comment:
indeed, you could log that the key is unknown and therefore skipped

Reply:
What you suggest is not what we want : we're reading a **generaldata.ini** file, which (as you know) contains many properties. 
But here, we want to extract some of them. 
If we return **false** as soon we meet a property we don't want to read, a lot of warnings will be raised.

Reply:
ok